feat(cli): add skip-cleanup parameter

umutcanbolat opened the pull request:
Introduces `skip-cleanup` param to be able to skip the initial cleanup.
The default value is false.
resolves #250

umutcanbolat commented:
I recommend reviewing the changes in vscode. Because GitHub diff makes it look like I changed the whole index file :D

umutcanbolat commented:
I recommend hiding the whitespace changes while reviewing. Otherwise, the change of indentation makes it look like the whole file is edited.
![image](https://user-images.githubusercontent.com/10065235/97097863-7cc26a00-1676-11eb-8b1d-99b458bca8d1.png)

Nargonath reviewed (changes requested):
Your PR needs to be updated since I merged several PR on master. Plus I'm not convinced about migrating to async/await here. I feel it makes the codebase harder to read and lots of the code is synchronous so it doesn't make sense to me to wrap everything in a big async function.

Nargonath commented:
It seems that my other comment on the code didn't go through. I had a question: what were the reasons behind the rewrite to async/await?

That's something I'd be happy to discuss but rather on its own PR and not mixed up with the feature you initially suggested.

umutcanbolat commented:
Hi, 

I think using async-await here makes it more readable. But we can consider this in another PR as you suggested.

So, with Promise chaining, it would look like this:

```js
Promise.resolve()
  .then(() => skipCleanup || fs.emptyDir(paths.appBuild))
  .then(() => {
    return new Promise((resolve, reject) => {
        // webpack related code here
        return resolve();
    });
  })
  .then(() => copyPublicFolder())
  .then(() => runHook('after initial build hook', spinner, afterInitialBuildHook));
```

With async-await, could be like this:

```js
skipCleanup || (await fs.emptyDir(paths.appBuild));

await new Promise((resolve, reject) => {
    // webpack related code here
    return resolve();
});

await copyPublicFolder();

runHook('after initial build hook', spinner, afterInitialBuildHook)
```

umutcanbolat commented:
I didn't have time this weekend, but I will resolve conflicts and do the changes during the next week.
Best!

umutcanbolat commented:
Hi again. Rebased into master and converted the async-await logic back to the promise chaining as you asked to discuss this in a separate PR.
Have a nice week!

Nargonath commented:
Hi @umutcanbolat! Thanks for updating your PR, I should be able to have a look at it during the week. No worries about the previous week-end. There is no rush. 😉

umutcanbolat commented:
Hi! Any update on this?
